mlir-sycl/test/Transforms/constant-propagation.mlir

```diff
@@ -1162,3 +1191,201 @@ llvm.func internal @foo_constant_constant_global_local_size(
          !llvm.ptr, !llvm.ptr, !llvm.ptr, !llvm.ptr) -> ()
   llvm.return
 }
+
+// -----
+
+// COM: Third split
+
+// CHECK-LABEL: ConstantPropagationPass
+// CHECK-NEXT:   (S) 7 num-propagated-constants   - Number of propagated constants
+// CHECK-NEXT:   (S) 7 num-replaced-explicit-args - Number of replaced explicit arguments
+// CHECK-NEXT:   (S) 0 num-replaced-implicit-args - Number of replaced implicit arguments
+
+!sycl_array_1_ = !sycl.array<[1], (memref<1xi64>)>
+!sycl_id_1_ = !sycl.id<[1], (!sycl_array_1_)>
+!sycl_range_1_ = !sycl.range<[1], (!sycl_array_1_)>
+!sycl_accessor_1_i32_w_gb = !sycl.accessor<[1, i32, write, global_buffer], (!sycl.accessor_impl_device<[1], (!sycl_id_1_, !sycl_range_1_, !sycl_range_1_)>, !llvm.struct<(ptr<i32, 1>)>)>
+!sycl_accessor_host = !sycl.accessor<[1, i32, write, global_buffer], (!llvm.void)>
+!sycl_accessor_host_range = !sycl.accessor<[1, i32, write, global_buffer], (!sycl_range_1_)>
+!sycl_accessor_host_offset = !sycl.accessor<[1, i32, write, global_buffer], (!sycl_range_1_, !sycl_id_1_)>
+
+gpu.module @kernels {
+  func.func private @init(%acc: memref<1x!sycl_accessor_1_i32_w_gb>,
+                          %ptr: memref<?xi32, 1>,
+                          %accRange: memref<?x!sycl_range_1_>,
+                          %memRange: memref<?x!sycl_range_1_>,
+                          %offset: memref<?x!sycl_id_1_>)
+
+// COM: Constant access range and offset
+
+// CHECK-LABEL:     gpu.func @k0(
+// CHECK-SAME:                   %[[VAL_0:.*]]: memref<?xi32, 1>, %[[VAL_1:.*]]: memref<?x!sycl_range_1_>, %[[VAL_2:.*]]: memref<?x!sycl_range_1_>, %[[VAL_3:.*]]: memref<?x!sycl_id_1_>) kernel {
+// CHECK-NEXT:        %[[VAL_4:.*]] = arith.constant 512 : index
+// CHECK-NEXT:        %[[VAL_5:.*]] = sycl.range.constructor(%[[VAL_4]]) : (index) -> memref<1x!sycl_range_1_>
+// CHECK-NEXT:        %[[VAL_6:.*]] = memref.cast %[[VAL_5]] : memref<1x!sycl_range_1_> to memref<?x!sycl_range_1_>
+// CHECK-NEXT:        %[[VAL_7:.*]] = arith.constant 1 : index
+// CHECK-NEXT:        %[[VAL_8:.*]] = sycl.id.constructor(%[[VAL_7]]) : (index) -> memref<1x!sycl_id_1_>
+// CHECK-NEXT:        %[[VAL_9:.*]] = memref.cast %[[VAL_8]] : memref<1x!sycl_id_1_> to memref<?x!sycl_id_1_>
+// CHECK-NEXT:        %[[VAL_10:.*]] = memref.alloca() : memref<1x!sycl_accessor_1_i32_w_gb2>
+// CHECK-NEXT:        func.call @init(%[[VAL_10]], %[[VAL_0]], %[[VAL_6]], %[[VAL_2]], %[[VAL_9]]) : (memref<1x!sycl_accessor_1_i32_w_gb2>, memref<?xi32, 1>, memref<?x!sycl_range_1_>, memref<?x!sycl_range_1_>, memref<?x!sycl_id_1_>) -> ()
```

Comment:
> `// COM: Constant access range and offset`

This is still passing `%accRange` as 4th arg to `@init`.

Reply:
My bad, I was using wrong parameter order